proxy_agent/src/key_keeper.rs

```diff
@@ -97,18 +98,18 @@ fn poll_secure_channel_status(
         if !first_iteration {
             // skip the sleep for the first loop
 
-            let sleep =
-                if key_keeper_wrapper::get_current_secure_channel_state(shared_state.clone())
+            let sleep = if key_keeper_wrapper::get_current_secure_channel_state(shared_state.clone())
```

Comment:
Would it be worth breaking out the sleep duration calculation into it's own function, or at the very least breaking the conditions into their own variables to make the comparison itself more descriptive?
```rust
#[inline]
pub fn get_sleep_interval(shared_state: Arc<Mutex<SharedState>>, fallback_interval: Duration) -> Duration {
    let in_unknown_state = key_keeper_wrapper::get_current_secure_channel_state(shared_state.clone()) == UNKOWN_STATE;
    let fallback_interval_is_longer = interval > FREQUENT_PULL_INTERVAL;
    let has_not_timed_out = helpers::get_elapsed_time_in_millisec() < FREQUENT_PULL_TIMEOUT_IN_MILLISECONDS;
    
    if in_unknown_state && fallback_interval_is_longer && has_not_timed_out {
        return FREQUENT_PULL_INTERVAL;
    }
    return interval;
}

Reply:
it is the same thing since the if condition already breakout for multiple lines